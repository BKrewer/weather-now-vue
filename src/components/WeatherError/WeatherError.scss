.weather-error {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;

    &__message {
        color: $warning;
        font-weight: 500;
        margin-bottom: 16px;
    }

    &__reload {
        background-color: transparent;
        border-radius: 20px;
        border: 2px solid $dark-grey;
        color: $dark-grey;
        cursor: pointer;
        font-size: 16px;
        font-weight: 500;
        padding: 10px 16px;
    }
}