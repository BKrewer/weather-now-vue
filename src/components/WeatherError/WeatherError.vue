<template>
  <div class="weather-error">
    <span class="weather-error__message">Something went wrong</span>
    <button @click="emitNewRequest" class="weather-error__reload">Try Again</button>
  </div>
</template>

<script>
import bus from '../../utils/bus'

export default {
    methods: {
        emitNewRequest() {
          bus.emit('newRequest')
        }
    }
}
</script>

<style lang="scss" src="./WeatherError.scss" />