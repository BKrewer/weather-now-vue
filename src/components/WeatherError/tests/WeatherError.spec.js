import { shallowMount } from '@vue/test-utils'
import WeatherError from '@/components/WeatherError/WeatherError.vue'

describe('WeatherError.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(WeatherError)
    expect(wrapper.element).toMatchSnapshot();
  })
})