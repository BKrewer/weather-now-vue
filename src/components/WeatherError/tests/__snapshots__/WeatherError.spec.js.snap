// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherError.vue renders correctly 1`] = `
<div
  class="weather-error"
>
  <span
    class="weather-error__message"
  >
    Something went wrong
  </span>
   
  <button
    class="weather-error__reload"
  >
    Try Again
  </button>
</div>
`;
